ios/Components/InstantComponent.swift

```diff
@@ -0,0 +1,155 @@
+//
+// Copyright (c) 2022 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+
+import Adyen
+import Foundation
+import PassKit
+import React
+
+@objc(AdyenInstant)
+internal class InstantComponent: BaseModule {
+    
+    private var currentComponent: PresentableComponent?
+    private var currentPaymentComponent: PaymentComponent?
+    private var actionHandler: AdyenActionComponent?
+    
+    @objc
+    override static func requiresMainQueueSetup() -> Bool { true }
+    override func stopObserving() {}
+    override func startObserving() {}
+    override func supportedEvents() -> [String]! { super.supportedEvents() }
+
+    @objc
+    func hide(_ success: NSNumber, event: NSDictionary) {
+        DispatchQueue.main.async {
+            self.currentComponent?.finalizeIfNeeded(with: success.boolValue)
+            UIApplication.shared.keyWindow?.rootViewController?.dismiss(animated: true)
+
+            self.actionHandler = nil
+            self.currentComponent = nil
+            self.currentPaymentComponent = nil
+        }
+    }
+
+    @objc
+    func open(_ paymentMethods: NSDictionary, configuration: NSDictionary) {
+        let paymentMethod: PaymentMethod
+        guard let data = try? JSONSerialization.data(withJSONObject: paymentMethods, options: []) else {
+            return assertionFailure("InstantComponent: Can not deserialize paymentMethods")
+        }
+        
+        if let paymentMethods = try? JSONDecoder().decode(PaymentMethods.self, from: data),
+           let firstPaymentMethod = paymentMethods.regular.first {
+            paymentMethod = firstPaymentMethod
+        }
+        else if let jsonPaymentMethod = try? JSONDecoder().decode(AnyPaymentMethod.self, from: data),
+                let anyPaymentMethod = jsonPaymentMethod.value {
+            paymentMethod = anyPaymentMethod
+        }
+        else {
+            return assertionFailure("InstantComponent: Can not parse payment method.")
+        }
```

Comment:
also this code seems similar to `AdyenDropIn.open()`, can we put somewhere to be reusable?

Reply:
Good point 🤔